server/rating_service/rating_service.py

```diff
@@ -0,0 +1,365 @@
+        ratings = {}
+        for team in summary.teams:
+            for player_id in team.player_ids:
+                ratings[player_id] = await self._get_player_rating(
+                    player_id, summary.rating_type
+                )
+
+        return [
+            TeamRatingData(
+                team.outcome,
+                {player_id: ratings[player_id] for player_id in team.player_ids},
+            )
+            for team in summary.teams
+        ]
+
+    async def _get_player_rating(
+        self, player_id: int, rating_type: RatingType
+    ) -> Rating:
+        if self._rating_type_ids is None:
+            self._logger.warning(
+                "Tried to fetch player data before initializing service."
+            )
+            raise ServiceNotReadyError("RatingService not yet initialized.")
+
+        rating_type_id = self._rating_type_ids.get(rating_type.value)
+        if rating_type_id is None:
+            raise ValueError(f"Unknown rating type {rating_type}.")
+
+        async with self._db.acquire() as conn:
+            sql = select([leaderboard_rating]).where(
+                and_(
+                    leaderboard_rating.c.login_id == player_id,
+                    leaderboard_rating.c.leaderboard_id == rating_type_id,
+                )
+            )
+
+            result = await conn.execute(sql)
+            row = await result.fetchone()
+
+        if not row:
+            return await self._get_player_legacy_rating(player_id, rating_type)
+
+        return Rating(row["mean"], row["deviation"])
+
+    async def _get_player_legacy_rating(
+        self, player_id: int, rating_type: RatingType
+    ) -> Rating:
+        if rating_type is RatingType.GLOBAL:
+            table = legacy_global_rating
+            sql = select([table.c.mean, table.c.deviation, table.c.numGames]).where(
+                table.c.id == player_id
+            )
+        elif rating_type is RatingType.LADDER_1V1:
+            table = legacy_ladder1v1_rating
+            sql = select(
+                [table.c.mean, table.c.deviation, table.c.numGames, table.c.winGames]
+            ).where(table.c.id == player_id)
+        else:
+            raise ValueError(f"Unknown rating type {rating_type}.")
+
+        async with self._db.acquire() as conn:
+
+            result = await conn.execute(sql)
+            row = await result.fetchone()
+
+            if not row:
+                new_rating = await self._create_default_rating(
+                    conn, player_id, rating_type
+                )
+                return new_rating
+
+            if rating_type is RatingType.GLOBAL:
+                # The following query counts entries in game_player_stats,
+                # where the rating of the given player improved.
+                # If the concerned game was never rated, after_mean is NULL
+                # and the row is not counted.
+                count_wins_query = (
```

Comment:
Cool. Unfortunately getting accurate win counts is going to be quite a bit more complicated than this. First off we will need to check the game mode. Global games are only games played on the `faf` game mode AFAIK, and it gets worse:

Way back when ladder was first introduced, there was no `ladder1v1` game mode, so all ladder games were also played on the `faf` game mode. Granted, this only affects the few players who have been playing FAF for a very long time, but these are also the players who will be most upset if their ladder history is messed up. I ran into this issue when trying to fix the ladder win ratio for the first time, so I've already done some work to figure out how we can distinguish these old ladder games from custom games with few false positives. You might want to take a look at:
https://github.com/FAForever/db/blob/develop/migrations/V67__add-old-ladder-games-to-winratio.sql and
https://github.com/FAForever/db/blob/develop/migrations/V65__fix-ladder-win-ratio.sql
Given that the ladder win ratio's still aren't working correctly, I think we should also do this win counting for ladder too.

Another thing we might need to keep an eye on is performance. As I recall, those migrations for updating every player's game count both took several hours to complete. Doing things lazily on a per-player basis should help a lot, but maybe we should track the execution time of this query, and put some extra thought into making sure it does only the minimum amount of work necessary.

Reply:
Hm,... I still think it might be a good place to cut corners and just set the global game wins to half of the total global games, especially if the performance is a problem and keep the current ladder game win count. For players with a *lot* of global games that is expected anyways and hard to take offense at, while for players with *very few* global games that care about it (aka that will play more games) it should converge somewhat quickly anyways.

Of course that means it wouldn't fix the ladder win count. But maybe that's something that's better done in a background job like the ones you described?
Since the counts are only kept in the `leaderboard_rating` table but not in the journal, we wouldn't make fixing these counts any harder I think, since we wouldn't need to rewrite the journal history.

Reply:
As for making these fixes a bit easier, maybe we could ask the community / the high level players, whether they would mind if their game counts would only reflect games from 2015 on (so that the mod would correctly distinguish between ladder and global).